### PR TITLE
PYSVC-755 Request Tracing Overhaul

### PR DESCRIPTION
**Pull request by Pandrex247**

### CODE REVIEW

**Comment by mikecroft** [comment]
The image scaling needs fixing - it's not possible to read the new image

File: documentation/payara-server/request-tracing-service/asadmin-commands.adoc [hunk 1]
```diff
@@ -181,92 +129,168 @@ asadmin> get-requesttracing-configuration
...
+|`--historicTraces` |Boolean |Whether to get the traces from the historic trace store or not |True |No
 |=======================================================================
 
 [[example-4]]
 === Example
 
 [source, shell]
 ----
-asadmin> get-requesttracing-configuration
+asadmin> list-requesttraces --historic false
```
Comment (OndrejM) [inline documentation/payara-server/request-tracing-service/asadmin-commands.adoc]: This example uses `--historic` which doesn't correspond to the docs above which mentions `--historicTraces`. Which one is correct?

File: documentation/payara-server/request-tracing-service/configuration.adoc [hunk 2]
```diff
@@ -19,20 +19,22 @@ restart the domain) to switch the request tracing service on.
...
+**Time Value**:: This defines the period of time that the adaptive sampling will try to hit the _Target Count_ in.
+**Time Unit**:: This defines the time unit of the _Time Value_ attribute.
+**Applications Only**:: When enabled, traces that originate from outside of deployed applications (e.g. the Admin Console) will not be sampled.
+**Threshold Value**:: This defines the number of units beyond which a request will be traced.
+**Threshold Unit**:: This defines the time duration per unit. The accepted options are any valid
+https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/TimeUnit.html[java.util.concurrent.TimeUnit] values. The default value is _SECONDS_.
+**Sample Rate First**:: When enabled, the sample rate will be checked as a trace begins, meaning that traces that would have broken the thresholds are potentially discarded. When disabled, the sample rate is checked after a trace has completed, meaning that you're only sampling traces that have have breached your threshold values. Having this disabled will potentially give more accurate trace data, at the expense of performance.
```
Comment (OndrejM) [inline documentation/payara-server/request-tracing-service/configuration.adoc]: A typo: "that have have breached"

File: documentation/payara-server/request-tracing-service/request-tracing-service.adoc [hunk 3]
```diff
@@ -25,15 +23,10 @@ The following types of requests are traced if the service is enabled:
...
-* Start and end of the request
-* Entering or exiting an EJB method execution
-* Entering or exiting a CDI bean method (only configured methods are
-traced)
-* SQL statement invocations via JDBC calls
-* Starting or finishing JTA transactions
-* JASPIC authentication calls
+In Payara Server and Micro 5.181, the Request Tracing Service was given an overhaul - traces are now in an OpenTracing-style format using Spans, instead of the original event stream format. Practically, this means that instead of having an ordered sequence of time stamped events constituting a trace (with separate entered and exited events), you now have a set of related traces (now called spans) with individual durations and attributes.
+
+The OpenTracing specification, found https://github.com/opentracing/specification/blob/master/specification.md[here], gives a more details description of what Spans are and the general trace format.
```
Comment (OndrejM) [inline documentation/payara-server/request-tracing-service/request-tracing-service.adoc]: Grammar: "gives a more details description"
Should it be rather: "gives more details about" ?

**Pull request merged**